crates/iota/src/client_commands.rs

```diff
@@ -2424,14 +2383,16 @@ impl IotaClientCommandResult {
         }
     }
 
-    pub async fn prerender_clever_errors(mut self, read_api: &ReadApi) -> Self {
+    pub async fn prerender_clever_errors(mut self, context: &mut WalletContext) -> Self {
         match &mut self {
             IotaClientCommandResult::DryRun(DryRunTransactionBlockResponse { effects, .. })
             | IotaClientCommandResult::TransactionBlock(IotaTransactionBlockResponse {
                 effects: Some(effects),
                 ..
-            }) => prerender_clever_errors(effects, read_api).await,
-
+            }) => {
+                let client = context.get_client().await.expect("Cannot connect to RPC");
+                prerender_clever_errors(effects, client.read_api()).await
```

Comment:
Perhaps we could have two ways to render, one with a connection and one without?

Reply:
Closing this as I prefer to keep this PR in line with upstream. Please open an issue or subsequent PR if you want to pursue the idea.